devtools/maven/src/main/java/io/quarkus/maven/Deployer.java

```diff
@@ -79,7 +79,7 @@ public static Set<String> getProjectDeployer(MavenProject project) {
         return project.getDependencies().stream()
                 .filter(d -> QUARKUS_GROUP_ID.equals(d.getGroupId()))
                 .map(d -> strip(d.getArtifactId()))
-                .filter(n -> Arrays.stream(Deployer.values()).anyMatch(e -> false))
+                .filter(n -> Arrays.stream(Deployer.values()).anyMatch(e -> e.extension.equals(n)))
```

Comment:
This is more complex than that. At the minimum, the `.map(d -> strip(d.getArtifactId()))` needs to be removed.

I created https://github.com/quarkusio/quarkus/issues/36325 for @iocanel to have a closer look as I'm not entirely sure what's the right fix.

I would recommend to just revert the changes related to that so that we can get this merged as it might take some time for Ioannis to have a closer look at this.

Reply:
@gsmet  I made revert, to keep the existing behaviour on the last commit